internal/converter/internal/otelcolconvert/otelcolconvert.go

```diff
@@ -246,6 +259,8 @@ func buildConverterTable() map[converterKey]componentConverter {
 			table[converterKey{Kind: component.KindExporter, Type: fact.Type()}] = conv
 		case connector.Factory:
 			table[converterKey{Kind: component.KindConnector, Type: fact.Type()}] = conv
+			// We need this so the connector is available as a destination for state.Next
```

Comment:
Connectors _must_ always act both as a receiver and as an exporter within a pipeline, otherwise it's not a valid OpenTelemetry collector configuration.

I'll try to make it more apparent.